src/storage/disk_table.cc

```diff
@@ -1097,8 +1097,128 @@ void DiskTableRowIterator::SeekToFirst() {
 }
 inline bool DiskTableRowIterator::IsSeekable() const { return true; }
 
+bool DiskTable::AddIndex(const ::openmldb::common::ColumnKey& column_key) {
+    // TODO(denglong): support ttl type and merge index
+    auto table_meta = GetTableMeta();
+    auto new_table_meta = std::make_shared<::openmldb::api::TableMeta>(*table_meta);
+    std::shared_ptr<IndexDef> index_def = GetIndex(column_key.index_name());
+    if (index_def) {
+        if (index_def->GetStatus() != IndexStatus::kDeleted) {
+            PDLOG(WARNING, "index %s is exist. tid %u pid %u", column_key.index_name().c_str(), id_, pid_);
+            return false;
+        }
+        new_table_meta->mutable_column_key(index_def->GetId())->CopyFrom(column_key);
+    } else {
+        ::openmldb::common::ColumnKey* added_column_key = new_table_meta->add_column_key();
+        added_column_key->CopyFrom(column_key);
+    }
+    if (!index_def) {
+        auto cols = GetSchema();
+        if (!cols) {
+            return false;
+        }
+        std::map<std::string, ColumnDef> schema;
+        for (int idx = 0; idx < cols->size(); idx++) {
+            const auto& col = cols->Get(idx);
+            schema.emplace(col.name(), ColumnDef(col.name(), idx, col.data_type(), col.not_null()));
+        }
+
+        std::vector<ColumnDef> col_vec;
+        for (const auto& col_name : column_key.col_name()) {
+            auto it = schema.find(col_name);
+            if (it == schema.end()) {
+                PDLOG(WARNING, "not found col_name[%s]. tid %u pid %u", col_name.c_str(), id_, pid_);
+                return false;
+            }
+            col_vec.push_back(it->second);
+        }
+
+        uint32_t inner_id = table_index_.GetAllInnerIndex()->size();
```

Comment:
@dl239 do we need to check if there is any existing inner index whose `column_key` is the same as the new index? If so, we can merge it with one `InnerIndex`.

Reply:
In memory table, index in the same `InnerIndex` is in the same segment. If there are multi ts columns in one segment, the value of node in  first level `skiplist` is an array.  we cannot allocate more memory for reserved pos. This is different with `disktable`.  I think we can merge it in `disktable` if there are benefits in the same column family.

Reply:
Got it. I will deal with it.

Reply:
For disk table, we keep the same design as memory table. That is, do not merge the new index with same `column_key` with other existing ones. If doing the merge, we have to update all the existing `key-value` pairs in rocksdb as the `key` format may change from `pk+ts` to `pk+ts+ts_col`.

Reply:
> For disk table, we keep the same design as memory table. That is, do not merge the new index with same `column_key` with other existing ones. If doing the merge, we have to update all the existing `key-value` pairs in rocksdb as the `key` format may change from `pk+ts` to `pk+ts+ts_col`.

Thank you for the reminder. 
I find the code that may be causing this error.
https://github.com/4paradigm/OpenMLDB/blob/fee7fa6dfd36dbe60c6868b74faadabc179e1062/src/storage/disk_table.cc#L255-L259